languages/python/reference/exercise-concepts/markdown.md

```diff
@@ -0,0 +1,48 @@
+# Concepts for markdown
+
+[Example Implementation](https://github.com/exercism/python/blob/master/exercises/markdown/example.py)
+
+## General  
+
+
+- functions: This exercise requires a re-write of pre-esiting code that uses functions to parse passed-in text in _markdown_.  Tests expect a function named `parse` that can be called to transform the _markdown_ formatted text and return HTML formatted text.
+
+- function arguments:  The example solution uses multiple functions that take various function arguments to operate on different pieces of the passed in markdown string.
+
+- `re` module:  Both the original code to be refactored for this exercise and the example solution import and use the `re` module for Regular Expressions in python.
+
+- `import` key word/ importing:  Both the original code to be refactored for the exercise and the example solution use the `import` keyword to import the `re` module in support of Regular Expressions in python.
+
+- `str.format()` :  The example solution uses `str.split()` to break the passed in markdown string into a list of lines broken up by the `\n` character.
+
+- `re.match()`:  Method of the `re` class that returns a `match` object with any matched values from a specified Regualr Expression or pre-compliled Regular Expression.  The example uses `re.match()` in multiple places to search for text patterns that need re-formatting or subsitituting. 
+
+- regular expressions:  As a seperate language with a seperate interpreter that is often embeded inside another programming language.  Like many other programming lanagues in use, python supports the Regular Expression language and implements a Regular Expression interpreter in the core `re` module.  Both the original code to be refactored for this exercise and the example solution use the core `re` module to access Regualr Expression functionality.
+
+- `return` keyword:  Most of the functions in the example solution specify a _return_ value using the `return` keyword.
+
+- `None`:  None (null) type.  Used to define a null variable or an object.  It is in of itself a singleton in any given python program.  The example uses `None` as an expecit return value in several functions.
```

Comment:
Spelling: "expecit" -> "explicit", and "is in of itself" should be "is in and of itself" or just "is itself". Also, while None is indeed Python's null type, it's a bit odd to say it _defines_ a null variable or object when you then go on to explain that None is a singleton ... any name that refers to None is None, and the only object that can be None is None, so all you really can do with None is assign a given name to refer to None ... it cannot really be used to define any other thing. Which is part of what you never need to explicitly `return None` when `return` does the same.

Reply:
> - `None`:  Pythons null type, referred to when a null or "placeholder" is needed.  It is in and of itself a singleton in any given python program.